### ISSUE TITLE
[Bug]: MEMBER_ACCOUNT_PAYMENT_INSTRUMENT_REQUIRED when updating aws_organizations_account resource

### ISSUE THREAD

**Issue by k-drumz** [hashicorp/terraform-provider-aws#26999]
### Terraform Core Version

1.2.3

### AWS Provider Version

3.75.2

### Affected Resource(s)

resource "aws_organizations_account" "account" {
  name      = local.friendly_name
  email     = azuread_group.account_email_group.mail
  parent_id = var.parent_id

  tags = {
    "Environment" = lower(var.environment)
    "env"         = lower(var.environment)
  }
}

### Expected Behavior

To be able to update the ```name``` and ```email``` attributes on already created accounts.

### Actual Behavior

When attempting to update the ```name``` and ```email``` attributes on already created accounts, the error is produced.

### Relevant Error/Panic Output Snippet

```shell
│ Error: ConstraintViolationException: The member account must be configured with a valid payment method, such as a credit card.
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "c0739716-c322-4d1f-9fb8-8346c368967e"
│   },
│   Message_: "The member account must be configured with a valid payment method, such as a credit card.",
│   Reason: "MEMBER_ACCOUNT_PAYMENT_INSTRUMENT_REQUIRED"
│ }
```

### Terraform Configuration Files

resource "aws_organizations_account" "account" {
  name      = local.friendly_name
  email     = azuread_group.account_email_group.mail
  parent_id = var.parent_id

  tags = {
    "Environment" = lower(var.environment)
    "env"         = lower(var.environment)
  }
}

### Steps to Reproduce

Terraform apply

### Would you like to implement a fix?

No

**Comment by justinretzolk** [comment]
Hey @k-drumz 👋 Thank you for taking the time to raise this! Because the [AWS Go SDK organizations package](https://docs.aws.amazon.com/sdk-for-go/api/service/organizations/) does not have a function to update the name or email address on an account, changes to either of these arguments force the resource to be destroyed and then re-created. There is an argument ([`close_on_deletion`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/organizations_account#close_on_deletion)) that determines what action should be taken when these deletion events occur. From the linked document:

> If true, a deletion event will close the account. Otherwise, it will only remove from the organization.

This argument defaults to `false`, so in your scenario, the accounts are only being removed from the organization. AWS' document on the [`RemoveAccountFromOrganization` API endpoint](https://docs.aws.amazon.com/organizations/latest/APIReference/API_RemoveAccountFromOrganization.html) then gives a clue on why you're receiving the error message:

> MEMBER_ACCOUNT_PAYMENT_INSTRUMENT_REQUIRED: To complete this operation with this member account, you first must associate a valid payment instrument, such as a credit card, with the account. Follow the steps at [To leave an organization when all required account information has not yet been provided](https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_accounts_remove.html#leave-without-all-info) in the AWS Organizations User Guide.

In this case, there seems to be two options to get past this error:

1. Associate a payment method with the accounts, and they will only be removed from the organization, rather than being closed.
2. Set the `close_on_deletion` argument to `true`, and the accounts will be closed, rather than only removed from the organization.

**Comment by nwhobart** [comment]
>     2. Set the `close_on_deletion` argument to `true`, and the accounts will be closed, rather than only removed from the organization.

I don't think this is the case. I'm running into the same problem and even setting `close_on_deletion` to `true` still outputs the same error.

```
╷
│ Error: error deleting AWS Organizations Account (123456789): ConstraintViolationException: The member account must be configured with a valid payment method, such as a credit card.
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "blah1"
│   },
│   Message_: "The member account must be configured with a valid payment method, such as a credit card.",
│   Reason: "MEMBER_ACCOUNT_PAYMENT_INSTRUMENT_REQUIRED"
│ }
│
│
```

**Comment by Denniskamau** [comment]
Running into the same issue, did you find a way to bypass this?

**Comment by ksemele-public** [comment]
Is this fixed somehow?
